### PR TITLE
Add a release post on Astro 0.18

### PR DESCRIPTION
**Pull request by matthewp**
This adds a release post for Astro 0.18

### CODE REVIEW

**Comment by matthewp** [comment]
Direct link to the post: https://astro-www-uzvmge3fh-pikapkg.vercel.app/blog/astro-018

**Comment by jasikpark** [comment]
Could you add `alt` text to the images or `role="presentation"` for any that don't add meaning and are just for presentation?

**Comment by jasikpark** [comment]
https://lighthouse-dot-webdotdevsite.appspot.com//lh/html?url=https%3A%2F%2Fastro-www-uzvmge3fh-pikapkg.vercel.app%2Fblog%2Fastro-018 👀

File: www/src/pages/blog/astro-018.astro [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+          ## Named slots
+
+          [Named slots](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots) are a standardized feature in shadow DOM and used in frameworks like [Vue](https://vuejs.org/v2/guide/components-slots.html#Named-Slots). In this release, Astro adds [named slots to Astro components](https://docs.astro.build/core-concepts/astro-components#slots), pages, and layouts.
+
+          Named slots are particularly useful for layout components. Layout components are what we call components that provide common HTML for a page, such as commonly added CSS in the &lt;head&gt; and the page's skeleton content.
+
+          Now, a Layout can specify slots for content to go into the "head" section, and some to go into the body's "content" section. The layout component defines its slots as part of its API. It can have as many or as few as you need!
+
+          Named slots will allow for **more robust **and dynamic Astro components. This will especially be useful for distributing components as packages where components take more options and are less opinionated.
```
Comment (jasikpark) [inline www/src/pages/blog/astro-018.astro:62]:
```suggestion
          Named slots will allow for **more robust** and dynamic Astro components. This will especially be useful for distributing components as packages where components take more options and are less opinionated.
```

Comment (jasikpark) [reply www/src/pages/blog/astro-018.astro:62]: Should "dynamic" be bold as well? Either way this is just a fix for the markdown rendering

File: www/src/pages/blog/astro-018.astro [hunk 2]
```diff
@@ -0,0 +1,112 @@
+---
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:1]: is there a reason that this couldn't be a markdown file, similar to `posts/introducing-astro.md`? It would be nice to stay consistent, and reuse the `BlogPost` layout which seems to be doing most of what's been copied here

File: www/src/pages/blog/astro-018.astro [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    </style>
+  </head>
+
+  <body>
+    <BlogHeader />
+    <BlogPost title={title} author={author} heroImage={heroImage} publishDate={publishDate}>
+        <Markdown>
+          After the success of Astro's launch we spent some time fixing bugs and taking user feedback on what to tackle next. Now some of the features that people requested are starting to get released.
+          
+          Astro 0.18 is the culmination and is the biggest release to date. This post highlights only some of what it brings.
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:36]: given how much is here, it would be good to do a bullet list summary similar to what Next.js does: https://nextjs.org/blog/next-11

**Comment by FredKSchott** [comment]
Agreed with the spacing comment @jasikpark left in Discord. If it helps, you can move the preview image to below the header instead of above it. Although, fixing the spacing should do the trick as well

File: www/src/pages/blog/astro-018.astro (hunk 2 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro:1]: There isn't a `posts/introducing-astro.md`. There's a posts `posts/introducing-astro.astro` which appears to not be linked to `blog/introducing-astro.astro` is what's linked.

Comment (jasikpark) [reply www/src/pages/blog/astro-018.astro:1]: Would it be possible to delete the /posts/introducing-astro.astro and add a vercel redirect to the /blog path?

File: www/src/pages/blog/astro-018.astro [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+          ![Code example that shows off using the new client:media hydrator.](/assets/blog/astro-018/responsive-hydration.png)
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro]: I'd recommend removing the `---` code fence, and adding an HTML 1-line comment explaining what's happening for someone quickly browsing. Something like:

```
<!-- Hydrate the sidebar on mobile devices, only! -->
<Sidebar client:media="(max-width: 700px)" />
```

File: www/src/pages/blog/astro-018.astro [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+          <!-- Saved from https://carbon.now.sh/16xchqPVdt5IEAY3Czg3 -->
+
+          ## Responsive component hydration
+
+          Astro uses [partial hydration](https://docs.astro.build/core-concepts/component-hydration), a technique to only include the client-side JavaScript for the parts of a page that are actually interactive. With Astro there were previously 3 strategies for partial hydration: hydrating on [page load](https://docs.astro.build/core-concepts/component-hydration#mycomponent-clientload-), when the CPU is [idle](https://docs.astro.build/core-concepts/component-hydration#mycomponent-clientidle-), or when the element is scrolled into [view](https://docs.astro.build/core-concepts/component-hydration#mycomponent-clientvisible-).
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:47]: this is a pretty heavy paragraph to put first, and doesn't really grab your attention. What about reordering to put the next paragraph first? Or even better, what if you merge this info into the third paragraph:

>  When developing responsive sites you often have components that are only visible on certain device sizes, like a mobile sidebar menu. Most developers don't have a good way to load mobile-only code on mobile devices, and end up wasting valuable performance loading code in the browser that will never even be seen by the end user.
> 
> Astro solves this problem using partial hydration, and v0.18 adds a new, 4th way to hydrate components using a **[CSS media query](https://docs.astro.build/core-concepts/component-hydration#mycomponent-clientmediaquery-)**. Add the `client:media` directive to a component and it will only hydrate once the media condition is met.

File: www/src/pages/blog/astro-018.astro [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+          This is all part of Astro's commitment to providing tools to minimize the amount of JavaScript you load.
+
+          ![Using multiple slots to provide the head and body parts of a Layout component.](/assets/blog/astro-018/named-slots.png)
+          <!-- https://carbon.now.sh/9UwJkMCezRIOhzac5VVp -->
+
+          ## Named slots
+
+          [Named slots](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots) are a standardized feature in shadow DOM and used in frameworks like [Vue](https://vuejs.org/v2/guide/components-slots.html#Named-Slots). In this release, Astro adds [named slots to Astro components](https://docs.astro.build/core-concepts/astro-components#slots), pages, and layouts.
+
+          Named slots are particularly useful for layout components. Layout components are what we call components that provide common HTML for a page, such as commonly added CSS in the &lt;head&gt; and the page's skeleton content.
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:62]: don't think you need this paragraph, since I believe the general concept of a layout is generally understood. 

you can just link the first reference to `Layout` in the next paragraph to our docs.

File: www/src/pages/blog/astro-018.astro [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+          Now, a Layout can specify slots for content to go into the "head" section, and some to go into the body's "content" section. The layout component defines its slots as part of its API. It can have as many or as few as you need!
+
+          Named slots will allow for **more robust** and dynamic Astro components. This will especially be useful for distributing components as packages where components take more options and are less opinionated.
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:66]: why bolded? consider removing

Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro:66]: this entire paragraph may be unnecessary, since it seems like a secondary benefit here that most readers won't connect with

File: www/src/pages/blog/astro-018.astro [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+          ![Solid.js logo](/assets/blog/astro-018/solid-logo.png)
+
+          ## Solid support
+
+          [Solid](https://www.solidjs.com/) is a JSX-based reactive framework that compiles to efficient DOM manipulation. Before Astro was even public, the Solid team expressed interest in building a renderer. With the new [Solid renderer](https://github.com/snowpackjs/astro/tree/main/packages/renderers/renderer-solid), now you can use Solid as a first-class framework in your project.
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:72]: "Before Astro was even public, the Solid team expressed interest in building a renderer."

Does this add anything to the paragraph? Feels unrelated, consider removing

File: www/src/pages/blog/astro-018.astro [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+          Even though Solid implements its own JSX compilation that differs from React and Preact, you're still able to use it side-by-side with any of those framework components. This is the promise of Astro, use all of your favorite framework(s)!
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:74]: "Even though Solid implements its own JSX compilation that differs from React and"

putting on my new user/reader hat: "wait, what? Is this something that I need to worry about? This sounds complicated, not easy..."

consider removing

File: www/src/pages/blog/astro-018.astro [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+          ![Lit logo](/assets/blog/astro-018/lit-logo.svg)
+
+          ## Lit support
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:78]: can this section be about Lit and custom elements? It's so much longer than the Solid section (for good reason!) so maybe expand the scope of the title a bit so that readers understand why

File: www/src/pages/blog/astro-018.astro [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+          * Fully rendered custom elements!
+          * Automatic inclusion of the declarative shadow DOM polyfill (for non-Chromium browsers).
+          * Integration with Lit's hydration system.
+
+          The Lit renderer is marked as experimental, but we're already excited to see that many popular libraries render correctly with it.  As other web component libraries adopt declarative shadow DOM and common APIs emerge for rendering, we hope to bring support to those libraries as well.
+
+
+          ![Example showing using the style global tag to provide global CSS classes for the entire page.](/assets/blog/astro-018/style-global.png)
+
+          ## &lt;style global&gt;
```
Comment (FredKSchott) [inline www/src/pages/blog/astro-018.astro:93]: this section and the one below it feel a bit light on content. Maybe combine into a single "Try out Astro v0.18" section that includes a bulleted list of "other cool stuff" and include `style global`, new docs site, and other highlights.

Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro:93]: Or, if this is mentioned in the list of features at the top of the site (see previous comment) you may not even need this section at all.

**Review by FredKSchott — COMMENTED** [review]
Looks good! Left some comments

File: www/src/pages/blog/astro-018.astro (hunk 2 above)
Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro:1]: Ah shoot, you're right I was looking at `examples/blog` and not `www/blog`. Nevermind!

Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro:1]: It would be good to move this into a markdown file though so that the two could share this layout logic, but that's less pressing.

Comment (matthewp) [reply www/src/pages/blog/astro-018.astro:1]: This one probably could be, but the other is using `<Note />` and a couple of other components so I think it can't be until we get `.mdc` files.

Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro:1]: yup, probably blocked by our "components in markdown" work

File: www/src/pages/blog/astro-018.astro (hunk 3 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro]: good idea, just added that.

File: www/src/pages/blog/astro-018.astro (hunk 2 above)
Comment (jasikpark) [reply www/src/pages/blog/astro-018.astro:1]: > Would it be possible to delete the /posts/introducing-astro.astro and add a vercel redirect to the /blog path?

I made a PR for this here: https://github.com/snowpackjs/astro/pull/876 - I might be missing something but I assume we don't want duplicated files for the blog?

File: www/src/pages/blog/astro-018.astro (hunk 8 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro]: Looks like you edited this out.

File: www/src/pages/blog/astro-018.astro [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+          * __[Responsive partial hydration:](#responsive-component-hydration)__ Hydrate components with CSS media queries.
+          * __[Named slots:](#named-slots)__ Support multiple content entrypoints inside of Astro components.
+          * __[Solid.js support:](#solid-support)__ Use [Solid.js](https://www.solidjs.com/) components in Astro.
+          * __[Lit support:](#solid-support)__ Use [Lit SSR](https://lit.dev/) to get server-side rendering for web components.
+          * [`<style global>` support](https://docs.astro.build/guides/styling#overview), [GitHub syntax highlighting](https://twitter.com/n_moore/status/1417881860051509250) and a [shiny new docs site.](https://docs.astro.build/)
+
+          ## Responsive partial hydration
+
+
+          <img src="/assets/blog/astro-018/responsive-hydration.png" alt="Code example that shows off using the new client:media hydrator." />
```
Comment (matthewp) [inline www/src/pages/blog/astro-018.astro:48]: I like this new image, but the component is for a DesktopSidebar and the media query is for < 700px! I'll update it. Curious how you got the file size down though.

File: www/src/pages/blog/astro-018.astro [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+
+          Adding support for Solid (our third officially-supported JSX framework) wasn't easy, and required a major refactoring of how Astro handles JSX. The final result was worth it: Astro is now much better at supporting different types of JSX and can even support the new React v17 JSX transform. This new foundation for Astro should support us well into the future.
+
+          The release of Solid v1.0 is the perfect showcase for Astro's multi-framework promise: try out your first Solid component in an existing project side-by-side with the rest of your components. Where other build tools force you to choose a single framework and stick with it, Astro makes multi-framework projects easy for short-term code migrations or long-term site architectures.
+
+
+          ## Lit SSR support
+
+          ![Lit logo](/assets/blog/astro-018/lit-element.jpeg)
```
Comment (matthewp) [inline www/src/pages/blog/astro-018.astro:80]: Is this an older logo? I can't find this one anywhere. Going to reset back to the newer Lit one, since that's what our renderer works with.

File: www/src/pages/blog/astro-018.astro [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+
+          [Solid](https://www.solidjs.com/) is a JSX-based UI framework that bills itself as the familiar, modern, more reactive alternative to React. We're really excited about Solid, and with the new [Solid renderer](https://github.com/snowpackjs/astro/tree/main/packages/renderers/renderer-solid) for Astro you can use Solid as a first-class framework in your project.
+
+          Adding support for Solid (our third officially-supported JSX framework) wasn't easy, and required a major refactoring of how Astro handles JSX. The final result was worth it: Astro is now much better at supporting different types of JSX and can even support the new React v17 JSX transform. This new foundation for Astro should support us well into the future.
+
+          The release of Solid v1.0 is the perfect showcase for Astro's multi-framework promise: try out your first Solid component in an existing project side-by-side with the rest of your components. Where other build tools force you to choose a single framework and stick with it, Astro makes multi-framework projects easy for short-term code migrations or long-term site architectures.
```
Comment (matthewp) [inline www/src/pages/blog/astro-018.astro:75]: Really like this paragraph.

File: www/src/pages/blog/astro-018.astro (hunk 7 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro]: Maybe it can be reworded? The idea I was trying to convey is that higher-level components can take advantage of named slots because they have less opinion on the *content*. So this change will help with building an Astro component ecosystem.

File: www/src/pages/blog/astro-018.astro (hunk 10 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro]: I like your changes here. I didn't want to make it be about "web components" in general because it's not true that we generally SSR web components, only Lit ones (and don't want to create an expectation to SSR all web components).

File: www/src/pages/blog/astro-018.astro (hunk 11 above)
Comment (matthewp) [reply www/src/pages/blog/astro-018.astro]: Noticed you removed this one (and the docs one). I don't think short sections are bad. The Next.js blog post you referenced had a lot of short sections.

File: www/src/pages/blog/astro-018.astro (hunk 13 above)
Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro]: yea I cheated: this was an image used on a web.dev blog post about lit-element. I couldn't find any good lit images at that dimensions (the single large lit square felt a bit too huge to me) so that was the best I could find.

File: www/src/pages/blog/astro-018.astro (hunk 12 above)
Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro]: Doh! I always confuse those two.

I used ImageOptim! @drwpow showed it to me a while back, it's great and simple.

File: www/src/pages/blog/astro-018.astro (hunk 7 above)
Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro]: I'm not opposed to adding something back here if its interesting to the reader. I originally recommended removal because it felt like one of those higher-level things that we care about as the people building the thing, but that most users/readers don't care about.

Maybe just boil it down to a single sentence?

File: www/src/pages/blog/astro-018.astro (hunk 11 above)
Comment (FredKSchott) [reply www/src/pages/blog/astro-018.astro]: Maybe remove the banner image then? It felt weird to me as a reader to have the title + section image take up more space than the content of the section :)
